### PR TITLE
chore(actions): add formatting check and auto-format PR jobs

### LINKED ISSUE #67: Add formatting check job to validation workflow
The job should run `spotless:check` maven goal, and block PR merge if it fails.
In addition, in case of failure, it should run `spotless:apply`, upload the `git diff` as job asset and comment the PR providing information about the errors.
If the diff is small, also inline it in the comment, otherwise only provide a link to the full diff file.
The comment should contain instructions to run `spotless:apply` locally.

### LINKED ISSUE #68: Add workflow to format PR based on comments
Investigate if it is possible and feasible to add a workflow triggered by comments on a PR to apply formatting and commit changes.

### PR DESCRIPTION
**Pull request by mcollovati**
A new job in validation workflow checks that formatting is ok, and if not, it prevents the PR from being merged.
If the code is not well formatted, the label `format:required` is applied to the PR and a comment reports the file that should be checked, a full diff and formatting instructions.
Otherwise, the `format:checked` label is applied.

Another workflow, triggered by the `auto-format` label, applies formatting to the PR and pushes the changes to the PR branch.
The push automatically triggers a new validation.

Closes #67
Closes #68

### CODE REVIEW

**Comment by mcollovati** [comment]
It can be tested by checking out #73, reverting the auto format commit and pushing changes.
The validation should fail because of wrong formatting.
Once all the checks are completed, apply the 'auto-format' label, and verify that the format job is executed, a new commit is pushed to the branch and the validation passes

File: .github/workflows/validation.yaml [hunk 1]
```diff
@@ -66,3 +70,125 @@ jobs:
...
+            const { owner, repo } = context.repo;
+            const prNumber = context.payload.pull_request.number;
+
+            await github.rest.issues.addLabels({
+              owner,
+              repo,
+              issue_number: prNumber,
+              labels: ['format:required'],
+            });
+      - name: Add format required label to PR
```
Comment (Dudeplayz) [inline .github/workflows/validation.yaml:176]:
```suggestion
      - name: Add format checked label to PR
```

File: .github/workflows/validation.yaml [hunk 2]
```diff
@@ -66,3 +70,125 @@ jobs:
...
+          echo "</details>" >> /tmp/message
+          echo "" >> /tmp/message
+          echo "" >> /tmp/message
+          echo "<details>" >> /tmp/message
+          echo "  <summary>Full Diff</summary>" >> /tmp/message
+          echo "" >> /tmp/message
+          echo '```patch' >> /tmp/message
+          cat /tmp/full-diff.txt  >> /tmp/message
+          echo '```' >> /tmp/message
+          echo "</details>" >> /tmp/message
```
Comment (Dudeplayz) [inline .github/workflows/validation.yaml:147]: Maybe use some template and do some replace operation on it?

File: .github/workflows/validation.yaml [hunk 3]
```diff
@@ -66,3 +70,125 @@ jobs:
...
+            
+            // Remove the label if it exists
+            const existingLabels = await github.rest.issues.listLabelsOnIssue({
+              owner,
+              repo,
+              issue_number: prNumber,
+            });
+            const labelsToRemove = existingLabels.data.map(label => label.name).filter( label => label.startsWith("format:") );
+            for (const labelToRemove of labelsToRemove) {
+              await github.rest.issues.removeLabel({
```
Comment (Dudeplayz) [inline .github/workflows/validation.yaml:94]: Is there also a .removeLabels operation, which directly takes all labels to remove, like for the .addLabels method?

File: .github/workflows/format-pr.yaml [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+            
+            // Remove the label if it exists
+            const existingLabels = await github.rest.issues.listLabelsOnIssue({
+              owner,
+              repo,
+              issue_number: prNumber,
+            });
+            const labelsToRemove = existingLabels.data.map(label => label.name).filter( label => label.startsWith("format:") );
+            for (const labelToRemove of labelsToRemove) {
+              await github.rest.issues.removeLabel({
```
Comment (Dudeplayz) [inline .github/workflows/format-pr.yaml:31]: Same question for .removeLabels method.

File: .github/workflows/validation.yaml (hunk 3 above)
Comment (mcollovati) [reply .github/workflows/validation.yaml:94]: There's only remove one, remove all or replace all

File: .github/workflows/validation.yaml (hunk 2 above)
Comment (mcollovati) [reply .github/workflows/validation.yaml:147]: Good point.
I'll not have much time to fix it next week. 
Perhaps it can be changed later on?

**Review by Dudeplayz — APPROVED** [review]

**Pull request merged**